pawn.cpp

```diff
@@ -0,0 +1,41 @@
+#include "pawn.h"
+
+
+Pawn::Pawn(int row, int column) : Piece(row, column)
+{
+    this->setColumn(column);
+    this->setRow(row);
+}
+
+
+/**
+  @inheritdoc
+*/
+
+
+void Pawn::makeMove(int diffRow, int diffColumn)
+{
+    bool ok = (this->isPawnMove(diffRow, diffColumn) & this->isValidMove(diffRow, diffColumn));
+
+    if(!ok)
+        return;
+
+    int currentRow = this->getRow();
+    int currentColumn = this->getColumn();
+
+    this->setRow(currentRow + diffRow);
+    this->setColumn(currentColumn + diffColumn);
+}
+
+bool Pawn::isPawnMove(int diffRow, int diffColumn)

```

Comment:
As modifica numele metodei ca e redundant (suntem deja in clasa Pawn), as pune `isMoveLegal`. Cum ti se pare?